mipengine/node/monetdb_interface/monet_db_connection.py

```diff
@@ -0,0 +1,92 @@
+from typing import List
+
+import pymonetdb
+
+from mipengine.common.node_catalog import node_catalog
+from mipengine.node.config.config_parser import config
+
+OCC_MAX_ATTEMPTS = 50
+
+
+class Singleton(type):
+    """
+    Copied from https://stackoverflow.com/questions/6760685/creating-a-singleton-in-python
+    """
+
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class MonetDB(metaclass=Singleton):
+    """
+    MonetDB is a Singleton class because we want it to be initialized at runtime.
+
+    If the connection is a public module variable, it will be initialized at import time
+    from Celery and all the Celery workers will use the same connection instance.
+
+    We want one MonetDB connection instance per Celery worker/process.
+
+    """
+
+    def __init__(self):
+        global_node = node_catalog.get_global_node()
```

Comment:
Why do we have a `get_global_node` and a `get_local_node` instead of a `get_node`?

I remember there was a reason but not the reason.

Reply:
This version of node_catalog was created for demo purposes.
There is a ticket [MIP-14](https://team-1617704806227.atlassian.net/browse/MIP-14)  to rework it.